Sources/ArcGISToolkit/FloorFilter/FloorFilterView.swift

```diff
@@ -0,0 +1,500 @@
+//
+// Copyright 2021 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+import Foundation
+
+protocol FloorFilterViewControllerDelegate {
+    // Updates the Floor Filter levels list with the site and facility that were selected from the prompt
+    func siteFacilityIsUpdated(viewModel: FloorFilterViewModel)
+}
+
+public class FloorFilterView: UIViewController, FloorFilterViewControllerDelegate {
+    
+    /// Public variables and functions accessible to the developer
+    public var selectedSite: AGSFloorSite? {
+        return viewModel.selectedSite
+    }
+    
+    public var selectedFacility: AGSFloorFacility? {
+        return viewModel.selectedFacility
+    }
+    
+    public var selectedLevel: AGSFloorLevel? {
+        return viewModel.selectedLevel
+    }
+    
+    /// Listener when a level is changed
+    public var onSelectedLevelChangedListener : (() -> Void)? = nil
+
+    /// Refresh the view with the new map
+    public func refresh(geoView: AGSGeoView?){
+        self.geoView = geoView
+        state = FloorFilterState.initiallyCollapsed
+        updateViewsVisibilityForState(state: state)
+    }
+    
+    /// Variables for styling the Floor Filter View
+    public var fontSize: CGFloat = 14.0
+    public var fontName: String = "Avenir"
+    public var selectionColor: UIColor = UIColor(hexString: "#C7EAFF")
+    public var backgroundColor: UIColor = UIColor(hexString: "#F2F2F7")
+    public var selectedTextColor: UIColor = UIColor(hexString: "#004874")
+    public var unselectedTextColor: UIColor = UIColor(hexString: "#323236")
+    
+    // Floor Filter UI Elements and Constraints
+    @IBOutlet var floorFilterView: UIView!
+    @IBOutlet weak var siteBtn: UIButton!
+    @IBOutlet weak var levelsTableView: UITableView!
+    @IBOutlet weak var closeBtn: UIButton!
+    @IBOutlet weak var floorFilterStackView: UIStackView!
+    @IBOutlet weak var closeBtnHeight: NSLayoutConstraint!
+    @IBOutlet weak var siteBtnHeight: NSLayoutConstraint!
+    @IBOutlet weak var tableViewHeight: NSLayoutConstraint!
+    @IBOutlet weak var closeBtnWidth: NSLayoutConstraint!
+    @IBOutlet weak var siteBtnWidth: NSLayoutConstraint!
+    @IBOutlet weak var levelCellWidth: NSLayoutConstraint!
+    
+    // Floor Filter view styles variables
+    // These are the default, but will get updated with values that are provided during initialization
+    private var buttonHeight: Int = 35
```

Comment:
Width and height properties are always set as CGFloat. Would be nice to follow that instead of setting it as Int and then casting it to CGFloat.